crates/bevy_gltf/src/label.rs

```diff
@@ -0,0 +1,165 @@
+                f.write_str(&format!("Mesh{mesh}/Primitive{primitive}"))
+            }
+            GltfAssetLabel::MorphTarget { mesh, primitive } => {
+                f.write_str(&format!("Mesh{mesh}/Primitive{primitive}/MorphTargets"))
+            }
+            GltfAssetLabel::Texture(index) => f.write_str(&format!("Texture{index}")),
+            GltfAssetLabel::Material {
+                index,
+                is_scale_inverted,
+            } => f.write_str(&format!(
+                "Material{index}{}",
+                if *is_scale_inverted {
+                    " (inverted)"
+                } else {
+                    ""
+                }
+            )),
+            GltfAssetLabel::DefaultMaterial => f.write_str("DefaultMaterial"),
+            GltfAssetLabel::Animation(index) => f.write_str(&format!("Animation{index}")),
+            GltfAssetLabel::Skin(index) => f.write_str(&format!("Skin{index}")),
+            GltfAssetLabel::InverseBindMatrices(index) => {
+                f.write_str(&format!("Skin{index}/InverseBindMatrices"))
+            }
+        }
+    }
+}
+
+impl GltfAssetLabel {
+    /// Add this label to an asset path
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// # use bevy_asset::prelude::*;
+    /// # use bevy_scene::prelude::*;
+    /// # use bevy_gltf::prelude::*;
+    ///
+    /// fn load_gltf_scene(asset_server: Res<AssetServer>) {
+    ///     let gltf_scene: Handle<Scene> = asset_server.load(GltfAssetLabel::Scene(0).from_asset("models/FlightHelmet/FlightHelmet.gltf"));
+    /// }
+    /// ```
+    pub fn from_asset(&self, path: impl Into<AssetPath<'static>>) -> AssetPath<'static> {
+        path.into().with_label(self.to_string())
+    }
+}
+
+impl From<(&gltf::Material<'_>, bool)> for GltfAssetLabel {
+    /// Returns the label for the `material`.
+    fn from((material, is_scale_inverted): (&gltf::Material, bool)) -> Self {
+        if let Some(index) = material.index() {
+            GltfAssetLabel::Material {
+                index,
+                is_scale_inverted,
+            }
+        } else {
+            GltfAssetLabel::DefaultMaterial
+        }
+    }
+}
+
+impl From<&gltf::Scene<'_>> for GltfAssetLabel {
+    /// Returns the label for the `scene`.
+    fn from(scene: &gltf::Scene) -> Self {
+        GltfAssetLabel::Scene(scene.index())
+    }
+}
+
+impl From<(&gltf::Skin<'_>, bool)> for GltfAssetLabel {
+    /// Return the label for the `skin` as [`GltfAssetLabel::Skin`] or [`GltfAssetLabel::InverseBindMatrices`].
+    fn from((skin, as_inverse_bind_matrices): (&gltf::Skin, bool)) -> GltfAssetLabel {
+        if as_inverse_bind_matrices {
+            GltfAssetLabel::InverseBindMatrices(skin.index())
+        } else {
+            GltfAssetLabel::Skin(skin.index())
+        }
+    }
+}
+
+impl From<&gltf::Texture<'_>> for GltfAssetLabel {
+    fn from(texture: &gltf::Texture) -> GltfAssetLabel {
+        GltfAssetLabel::Texture(texture.index())
+    }
+}
```

Comment:
Ah I didn't realize how some of these need an "inverse" flag. Sorry about that. I don't think we should go with this. Seeing something like `(material, true).into()` would be very confusing. It's not clear what's going on, what `true` corresponds to, etc. I think just making these regular free functions like `material_to_label(material: &gltf::Material, is_scale_inverted: bool)` is straight forward and clear.